internal/messagequeue/messagequeue.go

```diff
@@ -409,12 +412,12 @@ func (mq *MessageQueue) sendIfReady() {
 }
 
 func (mq *MessageQueue) sendMessage() {
-	err := mq.initializeSender()
+	sender, err := mq.initializeSender()
 	if err != nil {
-		log.Infof("cant open message sender to peer %s: %s", mq.p, err)
-		// TODO: cant connect, what now?
-		// TODO: should we stop using this connection and clear the want list
-		// to avoid using up memory?
+		// If we fail to initialize the sender, the networking layer will
+		// emit a Disconnect event and the MessageQueue will get cleaned up
+		log.Infof("Could not open message sender to peer %s: %s", mq.p, err)
+		mq.Shutdown()
```

Comment:
This is triggering a "close of closed channel" bug.

Reply:
Can we just let this happen automatically on disconnect?

Reply:
I think we want to stop the message queue immediately so it doesn't try to send any more messages